### DISCUSSION TITLE
More Flexible Shipping Implementation

### DISCUSSION THREAD

**Discussion by benhavo** [saleor/saleor#10828] [Ideas]
I'm building a retail ecommerce site for a mid-sized retailer in the US. The retailer ships packages with a wide variety of weights, and wants to use the most accurate shipping costs for transactions. The customer has chosen ShipStation as a third party solution for handling their shipping needs. The prices for each shipment are uniquely determined based on current rates, weight and dimensions of the products, and origin and destination addresses. As such, the shipping cost is too complex to rely on Saleor's shipping methodology. 

There is talk of ongoing work related to a ShipStation app for Saleor, but I don't see any repository for the app. Additionally, I'm not sure an app would allow full functionality with ShipStation rate calculation since updating the rates would apply to more than just a single Order. It seems like a good place for a feature enhancement or perhaps a new plugin.

I'd like the ability to assign a shipping cost to an order. There just doesn't seem to be a good solution or work-around with the current structure. I'm wondering if anyone else has encountered this issue, and if any solutions have been found along the way.

**Comment by timuric** [comment]
@benhavo please look into custom shipping method webhook https://docs.saleor.io/docs/3.x/developer/extending/apps/synchronous-webhooks/shipping-webhooks#list-external-shipping-methods

We would love to collaborate on a ShipStation integration if you would like to contribute to the open source integrations.

**Reply by benhavo** [reply]
Thanks so much @timuric - this is exactly what I was looking for. We're in a bit of a time crunch at the moment, so I'm not sure when exactly I'll get to this feature, but I'm fairly certain I will be working on a ShipStation app at some point given this capability. I'll reach out when I'm ready to begin work. Hopefully our timelines match up - it would be great to collaborate and contribute!
